Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import { PublishTarget, SkillManifestFile } from '@bfc/shared';
+import formatMessage from 'format-message';
+import { css, Dropdown, Icon, IDropdownOption, TextField, TooltipHost } from 'office-ui-fabric-react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { useRecoilValue } from 'recoil';
+
+import { botDisplayNameState, dispatcherState, settingsState, skillManifestsState } from '../../../../recoilModel';
+import { CreatePublishProfileDialog } from '../../../botProject/CreatePublishProfileDialog';
+import { iconStyle } from '../../../botProject/runtime-settings/style';
+import { ContentProps, VERSION_REGEX } from '../constants';
+
+const styles = {
+  container: css`
+    height: 350px;
+    overflow: auto;
+  `,
+};
+
+const onRenderLabel = (props) => {
+  return (
+    <div
+      style={{
+        display: 'flex',
+        flexDirection: 'row',
+        marginBottom: '5px',
+      }}
+    >
+      <div
+        style={{
+          marginRight: '5px',
+          fontWeight: 600,
+          fontSize: '14px',
+        }}
+      >
+        <div
+          style={{
+            paddingLeft: '5px',
+            paddingRight: '5px',
+          }}
+        >
+          {props.label}
+        </div>
+      </div>
+      <TooltipHost content={props.ariaLabel}>
+        <Icon iconName="Info" styles={iconStyle(props.required)} />
+      </TooltipHost>
+    </div>
+  );
+};
+
+export const getManifestId = (
+  botName: string,
+  skillManifests: SkillManifestFile[],
+  { content: { $schema } = {} }: Partial<SkillManifestFile>
+): string => {
+  const [version] = VERSION_REGEX.exec($schema) || [''];
+
+  let fileId = version ? `${botName}-${version.replace(/\./g, '-')}-manifest` : `${botName}-manifest`;
+  let i = -1;
+
+  while (skillManifests.some(({ id }) => id === fileId)) {
+    if (i < 0) {
```

Comment:
nit: just to be safe maybe add a bail out condition after max tries by setting a maxTry value?

Reply:
replace with `  while (skillManifests.some(({ id }) => id === fileId) && i < skillManifests.length) ` . using   `i < skillManifests.length` to ensure loop can be ended